Assignment2Q1Test.java

```diff
@@ -0,0 +1,63 @@
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+import static com.elifguler.Assignment2Q1.printAncestors;
+import static org.junit.jupiter.api.Assertions.*;
+
+class Assignment2Q1Test {
+    private BinaryTreeNode<Integer> n1, n2, n3, n4, n5, n6, n7, n8;
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+    private final ByteArrayOutputStream errContent = new ByteArrayOutputStream();
+
+    @BeforeEach
+    public void setUp() {
+        // Set up the tree
+        n1 = new BinaryTreeNode<>(1, null, null);
+        n2 = new BinaryTreeNode<>(5, null, null);
+        n3 = new BinaryTreeNode<>(3, n1, n2);
+        n4 = new BinaryTreeNode<>(14, null, null);
+        n5 = new BinaryTreeNode<>(9, n3, n4);
+        n6 = new BinaryTreeNode<>(19, null, null);
+        n7 = new BinaryTreeNode<>(18, null, n6);
+        n8 = new BinaryTreeNode<>(16, n5, n7);
+
+        // Set up streams
+        System.setOut(new PrintStream(outContent));
```

Comment:
Clever, but it is a bit of an overkill - it is more of an integration test, that checks whole system end to end. Typically though one would not test code by capturing stdout/stderr. Especially stderr will be broken by any debugging messages that one might add to the code in the future.

Integration tests are important, but they are usually used for larger systems where multiple parts cooperate in non-trivial ways. They also give only very high level information about failure that then needs to be tracked down to particular piece of code.
To get more fine-grained information it is better to start with unit tests, testing separate classes directly with assertions like: assertEquals(n8.findAncestors(18), Arrays.asList(...)). Testing in such isolation immediately points to a particular piece of code that might be broken.
Usual practice is to have one test class for each non-trivial implementation class.

Could you please refactor test code to verify directly implementation rather than output?